lib/src/command/login.dart

```diff
@@ -26,7 +26,12 @@ class LoginCommand extends PubCommand {
     final credentials = oauth2.loadCredentials(cache);
     if (credentials == null) {
       final userInfo = await _retrieveUserInfo();
-      log.message('You are now logged in as $userInfo');
+      if (userInfo == null) {
+        log.warning('Could not retrieve your user-details.\n'
+            'You might have to run `pub logout` to delete your credentials  and try again.');
```

Comment:
extra space between `credentials` and `and`.

Reply:
Also say `dart pub` or use the variable we have for the top-level name